src/renderer/components/editable-list/editable-list.tsx

```diff
@@ -0,0 +1,57 @@
+import "./editable-list.scss"
+
+import React from "React";
+import { Icon } from "../icon";
+import { Input } from "../input";
+import { observable } from "mobx";
+import { observer } from "mobx-react";
+
+export interface Props<T> {
+  items: T[],
+  add: (newItem: string) => void,
+  remove: (info: { oldItem: T, index: number }) => void,
+  placeholder?: string,
+
+  // An optional prop used to convert T to a displayable string
+  // defaults to `String`
+  display?: (item: T) => string,
+}
+
+@observer
+export class EditableList<T> extends React.Component<Props<T>> {
+  @observable currentNewItem = "";
+
+  render() {
+    const { items, add, remove, display = String, placeholder = "Add new item..." } = this.props;
```

Comment:
I would prefer to see defaults in `static defaultProps = {..}` as done in other places.